packages/react-devtools-shared/src/backend/views/Highlighter/Canvas.js

```diff
@@ -0,0 +1,370 @@
+        );
+        const tipBounds = getNestedBoundingClientRect(
+          this.tipBoundsWindow.document.documentElement,
+          this.window,
+        );
+
+        this.tip.updatePosition(
+          {
+            top: outerBox.top,
+            left: outerBox.left,
+            height: outerBox.bottom - outerBox.top,
+            width: outerBox.right - outerBox.left,
+          },
+          {
+            top: tipBounds.top + this.tipBoundsWindow.scrollY,
+            left: tipBounds.left + this.tipBoundsWindow.scrollX,
+            height: this.tipBoundsWindow.innerHeight,
+            width: this.tipBoundsWindow.innerWidth,
+          },
+        );
+      }
+    });
+  }
+
+  fillCanvas(context: CanvasRenderingContext2D, rect: Rect, dims: any): void {
+    const {height, left, top, width} = rect;
+
+    //fill margin
+    const fillLeftMargin = left - dims.marginLeft;
+    const fillTopMargin = top - dims.marginTop;
+    const fillWidthMargin = width + dims.marginLeft + dims.marginRight;
+    const fillHeightMargin = height + dims.marginTop + dims.marginBottom;
+
+    context.fillStyle = highlightStyles.margin;
+    context.fillRect(
+      fillLeftMargin,
+      fillTopMargin,
+      fillWidthMargin,
+      fillHeightMargin,
+    );
+
+    //fill padding
+    context.fillStyle = highlightStyles.padding;
+    context.clearRect(left, top, width, height);
+    context.fillRect(left, top, width, height);
+
+    //fill content
+    const fillLeft = left + dims.borderLeft + dims.paddingLeft;
+    const fillTop = top + dims.borderTop + dims.paddingTop;
+    const fillWidth =
+      width -
+      dims.borderLeft -
+      dims.borderRight -
+      dims.paddingLeft -
+      dims.paddingRight;
+    const fillHeight =
+      height -
+      dims.borderTop -
+      dims.borderBottom -
+      dims.paddingTop -
+      dims.paddingBottom;
+
+    context.fillStyle = highlightStyles.background;
+
+    context.clearRect(fillLeft, fillTop, fillWidth, fillHeight);
+    context.fillRect(fillLeft, fillTop, fillWidth, fillHeight);
+  }
+
+  remove() {
+    this.tip.remove();
+    this.destroy();
+  }
+
+  drawBorder(
+    context: CanvasRenderingContext2D,
+    rect: Rect,
+    color: string,
+  ): void {
+    const {height, left, top, width} = rect;
+
+    // outline
+    context.lineWidth = 1;
+    context.strokeStyle = OUTLINE_COLOR;
+
+    context.strokeRect(left - 1, top - 1, width + 2, height + 2);
+
+    // inset
+    context.lineWidth = 1;
+    context.strokeStyle = OUTLINE_COLOR;
+    context.strokeRect(left + 1, top + 1, width - 1, height - 1);
+    context.strokeStyle = color;
+
+    context.setLineDash([0]);
+
+    // border
+    context.lineWidth = 1;
+    context.strokeRect(left, top, width - 1, height - 1);
+
+    context.setLineDash([0]);
+  }
+
+  destroy(): void {
+    if (canvas !== null) {
+      if (canvas.parentNode != null) {
+        canvas.parentNode.removeChild(canvas);
+      }
+      canvas = null;
+    }
+  }
```

Comment:
I think you need to clean up the container DIV element too?
```suggestion
  destroy(): void {
    if (canvas !== null) {
      if (canvas.parentNode != null) {
        canvas.parentNode.removeChild(canvas);
      }
      canvas = null;
    }

    const doc = this.window.document;
    doc.body.removeChild(this.container);

    this.container = null;
  }
```